accel/opencl/ops_opencl.py

```diff
@@ -158,19 +190,20 @@ def image(self):
       assert len(self.shape) == 3 and self.shape[2] == 4, f"bad shape for image {self.shape}"
       self._image = CLImage(shape=(self.shape[1], self.shape[0]))
       if self._buf is not None:
-        assert prod(self.shape) == prod(self._image.cl.shape)*4
+        assert prod(self.shape) == prod(self._image.shape)*4
```

Comment:
I'm not sure this is always right, what if it's a buffer with a smaller backing buffer?

Reply:
Hmm, changed it to check `<=` against total area of the underlying (potentially tiled) image.